lib/dry/monads/validated.rb

```diff
@@ -138,6 +138,7 @@ class Invalid < Validated
         #
         # @return [Object]
         attr_reader :error
+        alias_method :value!, :error
```

Comment:
If I recall correctly, using `value!` in `Invalid` is semantically incorrect, thus the alias must be removed.

The thing is: `Invalid` and `Valid` contain values of different types, hence the different accessors.

Reply:
Yes, you are right. In terms of semantical correctness, it's not perfect. I would consider this as a quick fix. Unfortunately, the existing class interface doesn't have the proper methods for fixing the main issue in "monadic" manner (by chaining methods calls).
Alternatively, I could write some `if` statement, but it will be based either on the type or `respond_to?` that doesn't seem clean for me.
```ruby
def apply(val = Undefined)
  result = Undefined.default(val) { yield }
  if result.respond_to?(:value!)
    self
  else
    alt_map { |errors| errors + result.error }
  end
end
```
If you have any better idea, please letme know.